### PR TITLE
[Feature] Add ESLint to the project

### PR DESCRIPTION
**Pull request by rsilvr**
# Description

- Added ESLint config files and plugin as a dev dependency.
- It should be available in most IDEs with the use of some plugin or as CLI running the command: `eslint .` in the desired directory.
- I used the recommended preset and changed some rules from "error" to "warning" where I found to be too strict or inconvenient to the the current standards of the codebase.
- The root file is configured to Node/Common.js projects and I overrode it in specific projects 
- I ran the plugin in every Bruno project and fixed some minor issues like unused imports and variables.
- In some very few cases I found some undeclared variables that should be causing those lines to break (console.logs mostly).
- Below some issues where I didn't make any changes but I think could better analyzed @helloanoop:

**packages/bruno-electron/src/ipc/network/awsv4auth-helper.js**
```
12:7 error 'credentialsProvider' is not defined no-undef
15:7 error 'credentials' is not defined no-undef
15:27 error 'credentialsProvider' is not defined no-undef
16:27 error 'credentials' is not defined no-undef
17:31 error 'credentials' is not defined no-undef
18:28 error 'credentials' is not defined no-undef
```
It seems that all those variables are being declared globally but maybe they could be declared as simple `const`s?

**packages/bruno-app/src/utils/importers/postman-collection.js**
```
25:34  error 'text' is not defined
```

**packages/bruno-app/src/utils/common/index.js**
```
56:12  error    'JSON5' is not defined
```

These seem like bugs

### Contribution Checklist:

- [X] **The pull request only addresses one issue or adds one feature.**
- [X] **The pull request does not introduce any breaking changes**
- [X] **I have added screenshots or gifs to help explain the change if applicable.**
- [X] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [X] **Create an issue and link to the pull request.**

Closes: #1382 

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

### CODE REVIEW

File: packages/bruno-query/src/index.ts [hunk 1]
```diff
@@ -133,11 +133,12 @@ export function get(source: any, path: string, ...fns: PredicateOrMapper[]) {
       case token === '..':
       case token === '.':
         break;
-      case token === '?':
+      case token === '?': {
```
Comment (rsilvr) [inline packages/bruno-query/src/index.ts:136]: https://eslint.org/docs/latest/rules/no-case-declarations

File: packages/bruno-app/src/pages/_app.js [hunk 2]
```diff
@@ -43,8 +43,11 @@ function MyApp({ Component, pageProps }) {
 
   if (!window.ipcRenderer) {
     return (
-      <div class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 mx-10 my-10 rounded relative" role="alert">
```
Comment (rsilvr) [inline packages/bruno-app/src/pages/_app.js:46]: class -> className

File: packages/bruno-app/src/components/RequestPane/QueryEditor/index.js [hunk 3]
```diff
@@ -45,7 +45,6 @@ export default class QueryEditor extends React.Component {
       brunoVarInfo: {
         variables: getAllVariables(this.props.collection)
       },
-      theme: this.props.editorTheme || 'graphiql',
```
Comment (rsilvr) [inline packages/bruno-app/src/components/RequestPane/QueryEditor/index.js:48]: There is no `editorTheme` props being passed to this component on any of its occurrences

**Comment by helloanoop** [comment]
This is quite a significant change to the codebase, and given the current size of the PR backlog, I don't see us reviewing and merging this anytime soon.

We'll chart out an effort later to incrementally adopt a linter in the future.

**Pull request closed without merging**